tests/linked_list_test.c

```diff
@@ -0,0 +1,239 @@
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+#include <aws/common/linked_list.h>
+#include <aws_test_harness.h>
+
+struct int_value {
+    struct aws_linked_list_node node;
+    int value;
+};
+
+static int test_linked_list_order_push_back_pop_front() {
+    struct aws_linked_list list;
+
+    static size_t list_size = 4;
+    aws_linked_list_init(&list);
+
+    struct int_value first = (struct int_value){.value = 1};
+    struct int_value second = (struct int_value){.value = 2};
+    struct int_value third = (struct int_value){.value = 3};
+    struct int_value fourth = (struct int_value){.value = 4};
+
+    ASSERT_INT_EQUALS(0, aws_linked_list_length(&list), "List size should be 0.");
+
+    aws_linked_list_push_back(&list, (struct aws_linked_list_node *)&first);
+    aws_linked_list_push_back(&list, (struct aws_linked_list_node *)&second);
+    aws_linked_list_push_back(&list, (struct aws_linked_list_node *)&third);
+    aws_linked_list_push_back(&list, (struct aws_linked_list_node *)&fourth);
+
+    ASSERT_INT_EQUALS(list_size, aws_linked_list_length(&list), "List size should be %d.", (int)list_size);
+
+    struct int_value *item;
+    item = (struct int_value *)aws_linked_list_front(&list);
+    ASSERT_TRUE(item != 0, "List front() should have been successful.");
+    aws_linked_list_pop_front(&list);
+    ASSERT_INT_EQUALS(first.value, item->value, "Item should have been the first item.");
+    ASSERT_INT_EQUALS(list_size - 1, aws_linked_list_length(&list), "List size should be %d.", (int)list_size - 1);
+
+    item = (struct int_value *)aws_linked_list_front(&list);
+    ASSERT_TRUE(item != 0, "List front() should have been successful.");
+    aws_linked_list_pop_front(&list);
+    ASSERT_INT_EQUALS(second.value, item->value, "Item should have been the second item.");
+    ASSERT_INT_EQUALS(list_size - 2, aws_linked_list_length(&list), "List size should be %d.", (int)list_size - 2);
+
+    item = (struct int_value *)aws_linked_list_front(&list);
+    ASSERT_TRUE(item != 0, "List front() should have been successful.");
+    aws_linked_list_pop_front(&list);
+    ASSERT_INT_EQUALS(third.value, item->value, "Item should have been the third item.");
+    ASSERT_INT_EQUALS(list_size - 3, aws_linked_list_length(&list), "List size should be %d.", (int)list_size - 3);
+
+    item = (struct int_value *)aws_linked_list_front(&list);
+    ASSERT_TRUE(item != 0, "List front() should have been successful.");
+    aws_linked_list_pop_front(&list);
+    ASSERT_INT_EQUALS(fourth.value, item->value, "Item should have been the fourth item.");
+    ASSERT_INT_EQUALS(list_size - 4, aws_linked_list_length(&list), "List size should be %d.", (int)list_size - 4);
+
+    aws_linked_list_clean_up(&list);
+    RETURN_SUCCESS("%s() pass", "test_linked_list_order_push_back_pop_front");
+
+failure:
```

Comment:
This is still using the old style for unit tests, not the new AWS_TEST_CASE. in particular, you get rid of RETURN_SUCCESS and the failure goto.